File: 0228-Summary Ranges/0228-Summary Ranges.cpp
class Solution {
public:
    vector<string> summaryRanges(vector<int>& nums) {
        vector<string> result;
        int i = 0;
        for (int j = 0; j < nums.size(); ++j) {
            if (j + 1 < nums.size() && nums[j + 1] == nums[j] + 1) {
                continue;
            }
            
            if (i == j) {
                result.push_back(to_string(nums[i]));
            }
            else {
                result.push_back(to_string(nums[i]) + "->" + to_string(nums[j]));
            }
            i = j + 1;
        }
        
        return result;
    }
};

File: 0201-0300/0228-Summary Ranges/0228-Summary Ranges.py
class Solution:
    def summaryRanges(self, nums: List[int]) -> List[str]:
        i = 0
        result = []
        while i < len(nums):
            start = i
            while i + 1 < len(nums) and nums[i] + 1 == nums[i + 1]:
                i += 1
            if i == start:
                result.append(str(nums[start]))
            else:
                result.append(str(nums[start]) + '->' + str(nums[i]))
            i += 1
        return result
